executor/batch_point_get.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"context"
+
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/table/tables"
+	"github.com/pingcap/tidb/tablecodec"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/hack"
+)
+
+// BatchPointGetExec executes a bunch of point select queries.
+type BatchPointGetExec struct {
+	baseExecutor
+
+	tblInfo  *model.TableInfo
+	idxInfo  *model.IndexInfo
+	handles  []int64
+	idxVals  [][]types.Datum
+	startTS  uint64
+	snapshot kv.Snapshot
+	inited   bool
+	values   [][]byte
+	index    int
+}
+
+// Open implements the Executor interface.
+func (e *BatchPointGetExec) Open(context.Context) error {
+	return nil
+}
+
+// Close implements the Executor interface.
+func (e *BatchPointGetExec) Close() error {
+	return nil
+}
+
+// Next implements the Executor interface.
+func (e *BatchPointGetExec) Next(ctx context.Context, req *chunk.Chunk) error {
+	req.Reset()
+	if !e.inited {
+		if err := e.initialize(ctx); err != nil {
+			return err
+		}
+		e.inited = true
+	}
+	if e.index >= len(e.values) {
+		return nil
+	}
+	for !req.IsFull() && e.index < len(e.values) {
+		handle, val := e.handles[e.index], e.values[e.index]
+		err := decodeRowValToChunk(e.base(), e.tblInfo, handle, val, req)
```

Comment:
What if the query only need part of the columns? I don't find the projection operation.

Reply:
The `decodeRowValToChunk` only extracts the columns in the SELECT statement field list.